### PR TITLE
Improve initial drop target in collections when keyboard dragging

### PR DESCRIPTION
**Pull request by devongovett**
This changes the logic used to select the initial drop target within a collection when keyboard dragging. Previously, the initial drop target was the collection root if supported, or before the first item. This meant that if you were trying to insert into a very long list, you'd be taken to the top first and need to press the down arrow many times to get to the item you want to drop on. Similarly, when reordering a list, you'd select some items to move, and the insertion point would move all the way to the top of the list rather than being able to easily move by one from the initial location.

This changes the behavior so that:

* The initial drop target is after the previously focused item. This means you can scroll the list to where you want by navigating with the keyboard, and when dropping there, it will retain your position rather than starting from the top.
* If there is no focused item, then the initial drop target is the root, as before.
* The above rules change slightly when there is a selection, which I felt improved things when reordering multiple items in a list. 
  * If the last focused item is also the first selected item, then the initial drop target is before the first selected item. This way, if you select multiple items from bottom to top, we assume you want to move the items upward in the list, so the insertion point is placed before your selection.
  * Otherwise, if the last focused item is selected, then the initial drop target is after the last selected item. This way, if you select from top to bottom, the insertion point is placed after your selection.

Still need to write tests for this behavior, but wanted to get feedback on these proposed interactions.

### CODE REVIEW

**Comment by snowystinger** [comment]
Tried each of them out, it definitely feels like an improvement over before.
In long lists I think it's preferable to do it this way because you can always use the home key to get to the root if that's actually what you intended. It reduces the amount of automatic scrolling that happens too, and I think that's a good thing.

**Comment by majornista** [comment]
I agree with @snowystinger this behavior is a big improvement.

**Comment by devongovett** [comment]
Thanks for taking a look. Added tests. 😄

File: packages/@react-aria/dnd/test/useDroppableCollection.test.js [hunk 1]
```diff
@@ -770,6 +770,190 @@ describe('useDroppableCollection', () => {
...
+      userEvent.tab({shift: true});
+      expect(document.activeElement).toBe(draggable);
+
+      pressKey('Enter');
+      act(() => jest.runAllTimers());
+
+      expect(document.activeElement).toHaveAttribute('aria-label', 'Drop on Two');
+    });
+
+    it('should default to on the last selected item there is a selection and only dropping on items is allowed', () => {
```
Comment (snowystinger) [inline packages/@react-aria/dnd/test/useDroppableCollection.test.js:918]: english is a little hard to parse
maybe
```suggestion
it('should default to `on` the last selected item in the selection if only dropping on items is allowed', () => {
```

**Review by snowystinger — APPROVED** [review]
looks good, tiny bit to make tests more readable, but otherwise looks good

**Review by snowystinger — CHANGES_REQUESTED** [review]
fix 17 tests :)

**Review by snowystinger — APPROVED** [review]

**Review by LFDanLu — APPROVED** [review]
LGTM. I did have one thought: what do you think about updating the drop target's focused key when the user is cycling through the options while in dragging mode? I felt that it was kinda odd that a user could scroll through the list options, drop their drag target somewhere, but then when tabbing back to the drop target the list scroll back to the initial entry point rather than keeping the scroll position as is.

To reproduce:
1.  Go to drag-and-drop--droppable-grid-with-many-items story
2. Tab to the right list (drop list) and arrow down so "Item 1" is focused
3. Tab to left list and begin keyboard dragging on any of the items
4. Scroll down the right list and drop the item below "Item 13". The view should not scroll after the drop
5. Tab to the right list again. The view should jump back to "Item 1"

Still digging locally to see if this is a simple story change or something to do with the selectionManager 

EDIT: perhaps setting selectionManager.focusedKey in onDropTargetEnter?

**Comment by devongovett** [comment]
@LFDanLu I think #1721 is what you're looking for?

**Comment by LFDanLu** [comment]
@devongovett Partially yes, but there is still the case where the user might scroll through the droppable collection while in dragging mode but then cancel the drag operation (not as common a scenario admittedly, but perhaps the user realizes that they selected the wrong item to drag). In #1721 the focusedKey for the droppable collection wouldn't be updated and tabbing back into the droppable collection would cause the jump in scroll position.

**Comment by devongovett** [comment]
I suppose, but I guess that's normally the case with scrolling anyway? Like if you just used the scroll wheel to browse through items, that doesn't affect the focused item. If you tabbed in afterward, it would jump the scroll position back to the last focused item too.

**Review by dannify — APPROVED** [review]
